search/includes/classes/class-versioningcleanupjob.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace Automattic\VIP\Search;
+
+class VersioningCleanupJob {
+
+	/**
+	 * The name of the scheduled cron event to run the versioning cleanup.
+	 */
+	const CRON_EVENT_NAME = 'vip_search_versioning_cleanup';
+
+	const WEEK_IN_SECONDS = 7 * 24 * 60 * 60;
+
+	const SEARCH_ALERT_SLACK_CHAT = '#vip-go-es-alerts';
+
+	public function __construct( $indexables, $versioning ) {
+		$this->indexables = $indexables;
+		$this->versioning = $versioning;
+	}
+
+	/**
+	 * Initialize the job class.
+	 *
+	 * @access  public
+	 */
+	public function init() {
+		add_action( self::CRON_EVENT_NAME, [ $this, 'versioning_cleanup' ] );
+
+		$this->schedule_job();
```

Comment:
I believe it would be best to attempt/check to schedule the job on `admin_init` instead. Less frontend load for option checking / memcache fetches.

Reply:
Good call, this has been switched to `admin_init` or `wp_loaded` (for CLI).